crates/ruff_linter/src/rules/ruff/rules/unnecessary_cast_to_int.rs

```diff
@@ -0,0 +1,143 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{AlwaysFixableViolation, Applicability, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Arguments, Expr, ExprCall, ExprNumberLiteral, Number};
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::TextRange;
+
+/// ## What it does
+/// Checks for `int` conversions of values that are already integers.
+///
+/// ## Why is this bad?
+/// Such a conversion is unnecessary.
+///
+/// ## Known problems
+/// This rule is prone to false positives due to type inference limitations.
```

Comment:
We should add an example for the cases where this is the case. I would expect that the rule is correct in almost all cases

Reply:
That's good to know. 

We should extend the documentation and list all possible false positives and why they're false positives.

Reply:
Thanks

Reply:
I'd probably rewrite this too, to make it sound less intimidating. 

```
When values incorrectly override the `__round__`, `__ceil__`, `__floor__`,
or `__trunc__` operators such that they don't return an integer,
this rule may produce false positives.
```

@AlexWaygood I'm leaning towards marking the fix as safe even in those cases. It's correct that this *could* alter behavior but the problem isn't the fix but the non-API compatible override of the said dunder methods.

Reply:
Yes, if you've implemented the methods correctly, this should be safe. I don't think we need to worry too much about edge cases where the user has implemented the dunder method in an incorrect way.